File: Base-n.py
# Author : ysh
# 2024/01/21 Sun 22:14:02
# https://atcoder.jp/contests/abc192/tasks/abc192_d
a = int(input())
b = int(input())

def check(x:int) -> bool:
    last = 0
    for i in str(a):
        last = last * x + int(i)
        if last > b:
            return False
    return True

def ck(l: int,r: int) -> int:
    if l + 1 == r: return l;
    mid = (l + r) >> 1
    if check(mid): return ck(mid,r)
    else: return ck(l,mid)

if len(str(a)) == 1:
    if a <= b:
        print(1)
    else:
        print(0)
    quit()

print(ck(max([int(i) for i in str(a)]),10 ** 20) - max([int(i) for i in str(a)]))
quit()

File: Base-n.cpp
// Author : ysh
// 2024/01/21 Sun 22:10:08
// https://atcoder.jp/contests/abc192/tasks/abc192_d
// NF
#include<bits/stdc++.h>
using namespace std;
string a;
long long b;
inline bool check(int x) {

}
int ck(int l,int r) {
    if(l + 1 == r) return l;
    int mid = (l + r) >> 1;
    if(check(mid)) return ck(mid,r);
    return ck(l,mid);
}
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    cin>>a>>b;


}